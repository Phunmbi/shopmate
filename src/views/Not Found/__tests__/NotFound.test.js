import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import NotFoundContainer  from '../index';

describe('<ConnectedHomePage />', () => {
  const props = {};
  const initialState = {};
  const mockStore = configureStore();
  const store = mockStore(initialState);
  let wrapper;

  beforeEach(() => {
    wrapper =  mount(
      <Provider store={store}>
        <MemoryRouter>
          <NotFoundContainer {...props} />
        </MemoryRouter>
      </Provider>
    );
    localStorage.clear();
  });

  it('renders appropriately', () => {
    expect(wrapper).toMatchSnapshot();
  });
});