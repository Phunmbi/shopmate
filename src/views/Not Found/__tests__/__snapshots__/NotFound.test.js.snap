// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectedHomePage /> renders appropriately 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "lpsu7v",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "lpsu7v",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <NotFound>
        <div
          className="not-found"
        >
          <p>
            Let's get you back
            <Link
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                 Home 
              </a>
            </Link>
          </p>
        </div>
      </NotFound>
    </Router>
  </MemoryRouter>
</Provider>
`;
